polaris-core/src/main/java/org/apache/polaris/core/persistence/LocalPolarisMetaStoreManagerFactory.java

```diff
@@ -75,17 +76,20 @@ private void initializeForRealm(RealmContext realmContext) {
   }
 
   @Override
-  public synchronized Map<String, PolarisMetaStoreManager.PrincipalSecretsResult> bootstrapRealms(
-      List<String> realms) {
+  public Map<String, PolarisMetaStoreManager.PrincipalSecretsResult> bootstrapRealms(
+      List<String> realms, Function<String, PrincipalSecretsGenerator> rootSecretsPerRealm) {
```

Comment:
I like the `PrincipalSecretsGenerator`. Is it necessary to have it per realm, though? Can it take the realm as a method argument instead

Reply:
I'd probably have to add realm to `PolarisCallContext`.

From my POV, I wanted to avoid exposing realm to lower-level code for better abstraction.

In practice only one realm is normally bootstrapped at a time, so we can probably drop the realm parameter completely and use env. vars. like `POLARIS_BOOTSTRAP_ROOT_CLIENT_ID`. WDYT?

Reply:
I've reworked the generator code a bit. Hope it's clearer wrt realms now.